src/python/pants/util/process_handler.py

```diff
@@ -106,8 +106,8 @@ def join_and_get_output():
 
 def _tee(infile, outfile, return_function):
   accumulator = io.BytesIO()
-  for line in iter(infile.readline, ""):
+  for line in iter(infile.readline, b""):
```

Comment:
This was causing `iter` to never stop, because in Py3 it couldn't find the unicode "", only bytes "". Py2 doesn't care because it allows either or.

Thanks @illicitonion for the find!